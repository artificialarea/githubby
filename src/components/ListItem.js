// format Date into something human readable [^1]
function formatDate(date) {
  var monthNames = [
      "Jan", "Feb", "Mar",
      "Apr", "May", "Jun", "Jul",
      "Aug", "Sep", "Oct",
      "Nov", "Dec"
  ];

  var day = date.getDate();
  var monthIndex = date.getMonth();
  var year = date.getFullYear();

  return monthNames[monthIndex] + ' ' + day + ', ' + year;
}

function formatTime(date) {
  var hours = date.getHours();
  var mins = date.getMinutes();
  var secs = date.getSeconds();
  if (mins < 10) {
      mins = '0' + mins;
  }
  if (secs < 10) {
      secs = '0' + secs;
  }

  return hours + ':' + mins + ':' + secs;
}

export default function ListItem(props) {
  const { item } = props;

  let modifiedDate = '';
  let modifiedTime = '';

  if (item.commit.author.date) {
    modifiedDate = formatDate(new Date(item.commit.author.date));
    modifiedTime = formatTime(new Date(item.commit.author.date));
  }

  return (
    <li key={item.sha}>
      <a href={item.html_url} target="_blank">
        <p className="commit--message">{item.commit.message}</p>
      </a>
      <div className="commit--attribution">
        {item.author && 
          <img src={item.author.avatar_url} className="avatar" alt="avatar" /> 
        }
        <p>
          {item.author && 
            <span className="author">{item.author.login}</span>
          } 
          {' '}
          <span className="date">{modifiedDate}</span>
          {' '} 
          <span className="time">{modifiedTime}</span>
        </p>
      </div>
    </li>
  )
}

ListItem.defaultProps = {
  item: {},
} 


/* FOOTNOTES //////////////////////////////////////////

[^1]: Format Dates
format Date into something human readable
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date
- https://stackoverflow.com/questions/3552461/how-to-format-a-javascript-date
- perhaps reiterate with a npm date-fns package?

*/