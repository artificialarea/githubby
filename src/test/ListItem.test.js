import React from 'react';
import ReactDOM from 'react-dom';
import ListItem from '../components/ListItem';
import fauxData from './fauxData';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom';


describe('<ListItem />', () => {

  it('renders something', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <ListItem />
      </BrowserRouter>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders a ListItem', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <ListItem item={fauxData}/>
      </BrowserRouter>
    )
    // toJson(wrapper)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

})