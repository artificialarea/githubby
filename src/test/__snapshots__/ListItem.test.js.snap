// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders a ListItem 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ListItem
    item={
      Array [
        Object {
          "author": Object {
            "avatar_url": "https://avatars.githubusercontent.com/u/1991943?v=4",
            "login": "artificialarea",
          },
          "commit": Object {
            "author": Object {
              "date": "2020-08-17T23:01:44Z",
            },
            "message": "test commit message",
          },
          "sha": "c32035e6e979e0d83ceecda1e3d175d4449d73dd",
        },
        Object {
          "author": Object {
            "avatar_url": "https://avatars.githubusercontent.com/u/1991943?v=4",
            "login": "artificialarea",
          },
          "commit": Object {
            "author": Object {
              "date": "2020-08-18T23:01:44Z",
            },
            "message": "test commit message 2",
          },
          "sha": "c32035e6e979e0d83ceecda1e3d175d4449d73zz",
        },
      ]
    }
  />
</Router>
`;

exports[`<ListItem /> renders something 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ListItem
    item={Object {}}
  />
</Router>
`;
